packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/SimulationResource.java

```diff
@@ -141,4 +151,16 @@ public Response createFromSimulationResult(
 				.build();
 		}
 	}
+
+	@GET
+	@Path("/{jobId}/partial-result")
+	@Produces(MediaType.SERVER_SENT_EVENTS)
+	@SseElementType(MediaType.APPLICATION_JSON)
+	@Tag(name = "Stream partial/intermediate simulation result associated with run ID")
+	public Publisher<CalibrationIntermediateResults> stream(
+		@PathParam("jobId") final String jobId
+	) {
+		// return Multi.createFrom().publisher(events);
+		return Multi.createFrom().publisher(events).select().where(event -> event.getJobId == jobId);
+	}
```

Comment:
Is there ever a risk here where no `event.getJobId == jobId` evaluates to `true`? Will that return null or an exception - do we need to catch that and return a specific response in that case?

Reply:
No, that's the purpose of SSE, just listening to the endpoint like a broadcast. We might never receive anything at all.